core/src/main/java/org/apache/calcite/rel/metadata/RelMdUniqueKeys.java

```diff
@@ -246,8 +246,14 @@ private RelMdUniqueKeys() {}
 
   public Set<ImmutableBitSet> getUniqueKeys(Aggregate rel, RelMetadataQuery mq,
       boolean ignoreNulls) {
-    // group by keys form a unique key
-    return ImmutableSet.of(rel.getGroupSet());
+    if (Aggregate.isSimple(rel) || ignoreNulls) {
+      // group by keys form a unique key
+      return ImmutableSet.of(rel.getGroupSet());
```

Comment:
Since the group set is immutable, we can directly return rel.getGroupSet()?

Reply:
I don't think so. `rel.getGroupSet()` returns `ImmutableBitSet` while here we should return `Set<ImmutableBitSet>`

Reply:
I see. Thanks for the clarification.